internal/services/webhook_cache.go

```diff
@@ -0,0 +1,59 @@
+package services
+
+import (
+	"sync"
+)
+
+// Webhook represents a single webhook registration for a particular tokenId and signal name
+type Webhook struct {
+	URL       string
+	Condition string
+}
+
+// WebhookCache is an in-memory map of: tokenId -> map[signalName] -> []Webhook
+type WebhookCache struct {
```

Comment:
Note that this will only work with a single instance consumer. It's probably fine as long as you don't need to share state btw instances. If this is just populating from the database every so often, and the DB is your central source of truth, then this is probably fine (no need for eg. redis/valkey)